File: source/c/lis.c
// gcc -o lis lis.c
#include <stdio.h>

int main(void) {
    int arr[] = {10, 9, 2, 5, 3, 7, 101, 18};
    int n = 8;
    int dp[8];
    for (int i = 0; i < n; i++) dp[i] = 1;

    for (int i = 1; i < n; i++) {
        for (int j = 0; j < i; j++) {
            if (arr[j] < arr[i] && dp[j] + 1 > dp[i]) dp[i] = dp[j] + 1;
        }
    }

    int max_len = 0;
    for (int i = 0; i < n; i++) if (dp[i] > max_len) max_len = dp[i];

    printf("lis length: %d\n", max_len);
    return 0;
}

File: source/javascript/lis.js
// Longest Increasing Subsequence (LIS)
function lis(arr) {
    const n = arr.length;
    if (n === 0) return 0;
    const dp = new Array(n).fill(1);
    for (let i = 1; i < n; i++) {
        for (let j = 0; j < i; j++) {
            if (arr[j] < arr[i] && dp[j] + 1 > dp[i]) dp[i] = dp[j] + 1;
        }
    }
    return Math.max(...dp);
}

const arr = [10, 9, 2, 5, 3, 7, 101, 18];
console.log(`lis length: ${lis(arr)}`);
